src/app/invest/invest.component.ts

```diff
@@ -46,7 +46,7 @@ export class InvestComponent implements OnInit {
 
     getWalletBalance() {
         SpinnerUtil.showSpinner();
-        this.walletService.getUserWallet().subscribe(res => {
+        this.walletService.walletChange$.subscribe(res => {
```

Comment:
You didn't get me. You need to subscribe to `walletChange$` only in cases when you need to wait for changes from the other components. On invest component, I assume that you don't need to observe changes on the wallet because you won't change it from somewhere else in the application.